packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -33,11 +33,13 @@ function isSSR(options: undefined | boolean | { ssr: boolean }): boolean {
 
 /** Transform .astro files for Vite */
 export default function astro({ config, devServer }: AstroPluginOptions): vite.Plugin {
+  let platform: NodeJS.Platform;
   let viteTransform: TransformHook;
   return {
     name: '@astrojs/vite-plugin-astro',
     enforce: 'pre', // run transforms before other plugins can
     configResolved(resolvedConfig) {
+      platform = os.platform(); // TODO: remove macOS hack
```

Comment:
👍 on the TODO.

Would a _magical big fix_ be if Vite normalized all paths as URLs?

Reply:
It would! Or if not URLs, then some format that converts back into a URL easily.

This change is actually unrelated to Vite; it’s just improving a compiler patch (but found while testing Vite IDs in this file)